packages/cli/src/lib/guards.ts

```diff
@@ -0,0 +1,67 @@
+import type { BuildConfig, PrebuildOptions } from "@brandingbrand/code-cli-kit";
+
+import { withAction } from "./action";
+
+/**
+ * Defines an action and wraps it with logging functionality using the "@brandingbrand/code-cli-kit".
+ * This utility is designed to work with asynchronous actions.
+ *
+ * @param action - The asynchronous action to be performed.
+ * @param name - A descriptive name for the action, used in logging messages.
+ * @returns A Promise that resolves when the action completes successfully.
+ *
+ * @remarks
+ * This function is a wrapper around the "withLog" utility from "@brandingbrand/code-cli-kit".
+ * It adds logging capabilities to the provided action, allowing for better visibility into
+ * the execution of asynchronous tasks.
+ */
+export function defineAction(
+  action: () => Promise<void | string>,
+  name: string
+) {
+  return withAction(action, name);
+}
+
+export type Transformer<T> = {
+  file: string;
+  transforms: Array<T>;
```

Comment:
I meant this transforms array seems more like an implementation detail and doesn't need to be exposed on the type as we don't have any code working with this transform array outside the individual transformer implementations. I was using forEach vs reduce as examples to show it was an implementation detail.

Reply:
I agree that this would be an implementation detail on how to use the transforms, which is left up to the transformer function which is used as the entry point by the transformer action. But I do think it's important to keep a standard amongst all transformers. It's much easier from a contribution and reviewer perspective to keep this as much of a standard as possible.

From the perspective you provided which is perfectly valid, this would mean that transforms don't even need to exist - everything can technically be done just in the transformer function. The point of the transforms array is separation of concerns for each type of transform.